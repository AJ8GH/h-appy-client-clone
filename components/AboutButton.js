import React from 'react'
import { useNavigation } from '@react-navigation/native';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native'

export default function AboutButton() {
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={() => navigation.navigate('About')}>
        <Text style={styles.aboutText}>
          About this App
        </Text>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: '#819595',
    flex: 0.2,
    height: '8%',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  aboutText: {
    fontFamily: 'Courier',
    color: '#363946'
  },
})
