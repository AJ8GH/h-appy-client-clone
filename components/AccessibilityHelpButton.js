import React from 'react';
import {
  Text, TouchableOpacity, ActivityIndicator, Alert
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const AccessibilityHelpButton = () => {
  return (
    <TouchableOpacity
      style={{ paddingTop: 5 }}
      onPress={() => Alert.alert(
        "Accesibility",
        "Accessibility description",
      )}
    >
        <Ionicons name="help-circle-outline" size={20} color="#B1B6A6" style={{ height: 20, width: 20, textAlign: 'center' }} />
    </TouchableOpacity>
  );
};

export default AccessibilityHelpButton;