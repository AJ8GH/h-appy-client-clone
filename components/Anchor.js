import React from 'react';
import { TouchableOpacity } from 'react-native';
import * as Linking from 'expo-linking';

const handlePress = (props) => {
  Linking.openURL(props.href);
  props.onPress && props.onPress();
};

export default function Anchor(props) {
  return (
    <TouchableOpacity {...props} style={{ alignItems: 'center' }} onPress={() => handlePress(props)}>
      {props.children}
    </TouchableOpacity>
  );
}
