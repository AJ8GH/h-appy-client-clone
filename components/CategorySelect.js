import React, { useState } from "react";
import MultiSelect from "react-native-multiple-select";
import { View, StyleSheet } from "react-native";

const items = [
  { id: 1, name: "a" },
  { id: 2, name: "b" },
  { id: 3, name: "c" },
  { id: 4, name: "d" },
];

export default function CategorySelect() {

  function onSelectedItemsChange(selectedItems) {
    setSelectedItems(selectedItems);
    console.log(selectedItems);
  }

  return (
    <View style={styles.selectContainer}>
      <MultiSelect
        hideTags
        items={items}
        uniqueKey="name"
        onSelectedItemsChange={onSelectedItemsChange}
        selectedItems={selectedItems}
        selectText="Pick categories"
        searchInputPlaceholderText="Search categories..."
        onChangeInput={(text) => console.log(text)}
        tagRemoveIconColor="#CCC"
        tagBorderColor="#CCC"
        tagTextColor="#CCC"
        selectedItemTextColor="#CCC"
        selectedItemIconColor="#CCC"
        itemTextColor="#000"
        displayKey="name"
        searchInputStyle={{ color: "#CCC" }}
        // submitButtonColor="#48d22b"
        // submitButtonText="Submit"
      />
    </View>
  );
}



const styles = StyleSheet.create({
  selectContainer: {
    flex: 1,
    width: '80%',
    borderRadius: 10,
    // overflow: "hidden"
  },
})
