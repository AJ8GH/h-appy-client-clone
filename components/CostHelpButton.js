import React from 'react';
import {
  Text, TouchableOpacity, ActivityIndicator, Alert,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const CostHelpButton = () => (
  <TouchableOpacity
    style={{ paddingTop: 5 }}
    onPress={() => Alert.alert(
      'Cost',
      'Rating of Free to ££££, categorising activities by how much they will set you back.',
    )}
  >
    <Text>
      <Ionicons name="help-circle-outline" size={20} color="#B1B6A6" />
    </Text>
  </TouchableOpacity>
);

export default CostHelpButton;
