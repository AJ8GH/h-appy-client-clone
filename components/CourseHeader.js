import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default function CourseHeader() {
  return (
    <View styles={styles.courseHeaderContainer}>
      <Text style={styles.courseHeader}>Nibbles</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  courseHeader: {
    fontSize: 40,
    fontFamily: 'Didot',
  },
  courseHeaderContainer: {
    flex: 1,
    maxHeight: 50,
  },

});
