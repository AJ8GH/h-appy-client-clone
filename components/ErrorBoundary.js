import React from 'react';
import { Text } from 'react-native';

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    console.error(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return (<Text>Something went wrong</Text>);
    }

    return this.props.children;
  }
}
