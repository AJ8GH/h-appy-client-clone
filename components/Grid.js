import {
  Text, View, StyleSheet, FlatList, Dimensions, Platform,
} from 'react-native';
import React from 'react';

const windowWidth = Dimensions.get('window').width;
const numColumns = 2;
const gridWidth = windowWidth / numColumns - 30;
const gridHeight = gridWidth / 2;

export default function Grid(props) {
  const { activity } = props;

  if (!activity.categories) { activity.categories = []; }
  const categories = activity.categories.join('\n');
  const data = [
    { id: 'header0', value: 'Size' },
    { id: 'header1', value: 'Categories' },
    { id: 'size', value: activity.size },
    { id: 'categories', value: categories },
    { id: 'header2', value: 'Cost' },
    { id: 'header3', value: 'Acccessibility' },
    { id: 'cost', value: 'Â£ '.repeat(activity.cost) || 'Free :)' },
    { id: 'accessibility', value: `${activity.accessibility}/10` },
  ];

  return (
    <FlatList
      data={data}
      renderItem={({ item }) => (
        <View style={containerStyle(item)}>
          <Text style={[styles.item, itemStyle(item)]}>{item.value}</Text>
        </View>
      )}
      keyExtractor={(item) => item.id}
      numColumns={numColumns}
    />
  );
}

function containerStyle(item) {
  const itemContainer = {
    width: gridWidth,
    height: gridHeight,
  };

  if (item.id.includes('header')) {
    return {
      width: gridWidth,
      height: gridWidth / 4,
    };
  }
  return itemContainer;
}

function itemStyle(item) {
  const padding = {
    padding: '20%',
  };

  if (!item.id.includes('header')) {
    return padding;
  }
}

const styles = StyleSheet.create({
  itemContainer: {
    width: gridWidth,
    height: gridWidth,
    flexDirection: 'row',
    borderRadius: 10,
  },
  item: {
    flex: 1,
    margin: 1,
    fontSize: 18,
    backgroundColor: '#353746',
    textAlign: 'center',
    color: '#fff',
    borderRadius: 4,
    overflow: 'hidden',
    fontFamily: Platform.OS === 'android' ? 'Roboto' : 'Chalkduster',
  },
});
