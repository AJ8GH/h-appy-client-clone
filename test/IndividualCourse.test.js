import React from 'react';
import renderer from 'react-test-renderer';
import { CourseHeader } from '../components/CourseHeader';

test('renders correctly', () => {
  const tree = renderer.create(<CourseHeader />).toJSON();
  expect(tree.children).toHaveLength(1);
});
