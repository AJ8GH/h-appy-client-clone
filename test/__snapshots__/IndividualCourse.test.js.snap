// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualCourse renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "alignItems": "center",
      "backgroundColor": "#f8f9d4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#c7524a",
          "flex": 1,
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 60,
          "justifyContent": "center",
          "maxHeight": 60,
          "position": "absolute",
          "top": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontFamily": "Didot",
            "fontSize": 25,
            "fontWeight": "bold",
            "justifyContent": "center",
          }
        }
      >
        H-Appy
      </Text>
    </View>
  </View>
  <View>
    <View
      styles={
        Object {
          "flex": 1,
          "maxHeight": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Didot",
            "fontSize": 40,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingBottom": 30,
        "paddingTop": 10,
        "width": "80%",
      }
    }
  >
    <Text>
      this is where the full list of appetisers could go. a little mini menu. maybe also a blurb about what the appetiser category means in terms of activity size
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Back to the Main Menu
      </Text>
    </View>
  </View>
</View>
`;
